### ISSUE TITLE
package dependencies/loading

### ISSUE THREAD

**Issue by wburke24** [RHESSys/RHESSysIOinR#8]
currently there are many other packages used in rhessysIOinR, they're included in the imports in the description, but that doesn't attach the packages for use, currently you have to attach all the packages along with RHESSysIOinR, but we should be able to get around this.

The options are either explicitly referencing the other packages when the functions are used (ie dplyr::XXX), attaching specific functions for use throughout, or attaching the entire packages.  see the hadley wickum take on it here http://r-pkgs.had.co.nz/description.html

**Comment by ryanrbart** [comment]
A couple of things here. First, we should always include the :: when using functions from other packages. I've started doing this but have gone back through the RHESSysIOinR code. It will make it so the user doesn't have to attach the package, though they will still need to have it downloaded.

We can remove the packages that are not used (and possibly even some that are rarely used). I wasn't being selective when requiring them. 

We need to investigate how to get package dependencies downloaded. When one downloads from CRAN, the dependencies are automatically downloaded also. This doesn't happen with the current way I am loading RHESSysIOinR (opening  Rproj and pressing build). There might be a way around this though, just not sure yet.

**Comment by wburke24** [comment]
yeah on my version I've been going through and adding the package and ::.  one issue is the pipes (%>%) which is kinda a pain, so maybe people can just load tidyr?

**Comment by tungmilan** [comment]
@wburke24 : I thought the `%>%` is a part of the `magrittr` package? Keep in mind that using pipe can slow everything down 
https://twitter.com/thanhtungmilan/status/1039953450228887552

**Comment by wburke24** [comment]
@tungmilan Yes it is, though it's included in tidyr too. And ithat's a good point, and eventually it would be nice to rewrite those bits, but for now we just have to deal with them.
